New .cargo format forces me to remove .cargo to run `cargo generate-lockfile` in sources/

I recently tried to upgrade to Twoliter 0.8.0 where you changed the format of how you vendor dependencies into the .cargo folder to make it more robust against version changes. Inherently great! But sadly if you've built a kit or a variant that has resulted in a `.cargo` folder this gets picked up by cargo (I am unsure how to disable it):
https://internals.rust-lang.org/t/problems-of-cargo-config-files-and-possible-solutions/12987/14
https://github.com/rust-lang/cargo/issues/12738

And for me it creates a recursive loop that gives an error like this:
```
mikn@hal8999 ~/d/m/s/p/b/b/sources ((72abfc6e))> cargo generate-lockfile
error: failed to get `bottlerocket-defaults-helper` as a dependency of package `settings-defaults-metal-dev v0.3.0 (/home/mikn/devel/molnett/src/platform/base/bottlerocket/sources/settings-defaults/metal-dev)`

Caused by:
  failed to load source for dependency `bottlerocket-defaults-helper`

Caused by:
  Unable to update https://github.com/bottlerocket-os/bottlerocket-settings-sdk?tag=bottlerocket-defaults-helper-v0.1.1

Caused by:
  the source git+https://github.com/bottlerocket-os/bottlerocket-settings-sdk?tag=bottlerocket-defaults-helper-v0.1.1 requires a lock file to be present first before it can be
  used against vendored source code

  remove the source replacement configuration, generate a lock file, and then
  restore the source replacement configuration to continue the build
mikn@hal8999 ~/d/m/s/p/b/b/sources ((72abfc6e)) [101]> 
```

I can fix this by removing the .cargo folder, but I do need to do that every time which feels a bit crufty.

The secondary problem is that it is also a lot more sensitive to races, as now it seems to expect to be allowed to do more in this folder than it was previously. For now I am downgrading to 0.7.3 again.

Oh really, quick find!

If you are interested, the race I am encountering is this:
```
[2025-03-12T20:41:42Z DEBUG twoliter__bin::common] Running: Command { std: "cargo" "make" "--disable-check-for-updates" "--makefile" "/home/builder/.bazel/execroot/_main/platform/base/bottlerocket/build/tools/Makefile.toml" "--cwd" "/home/builder/.bazel/execroot/_main/platform/base/bottlerocket" "-e=TLPRIVATE_SDK_IMAGE=public.ecr.aws/bottlerocket/bottlerocket-sdk:v0.50.1" "-e=BUILDSYS_OUTPUT_GENERATION_ID=1" "-e=TWOLITER_TOOLS_DIR=/home/builder/.bazel/execroot/_main/platform/base/bottlerocket/build/tools" "-e=BUILDSYS_ARCH=x86_64" "-e=BUILDSYS_VARIANT=metal-dev" "-e=BUILDSYS_VERSION_IMAGE=1.2.0" "-e=GO_MODULES=" "-e=BUILDSYS_UPSTREAM_SOURCE_FALLBACK=false" "build", kill_on_drop: false }
2025-03-12 20:41:48 [ERROR] failed to fetch crates:     Blocking waiting for file lock on package cache
    Updating git repository `https://github.com/bottlerocket-os/bottlerocket-settings-sdk`
warning: spurious network error (3 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
error: failed to get `bottlerocket-defaults-helper` as a dependency of package `settings-defaults-metal-dev v0.3.0 (/tmp/sources/settings-defaults/metal-dev)`

Caused by:
  failed to load source for dependency `bottlerocket-defaults-helper`

Caused by:
  Unable to update https://github.com/bottlerocket-os/bottlerocket-settings-sdk?tag=bottlerocket-defaults-helper-v0.1.1#0cf85735

Caused by:
  failed to fetch into: /tmp/.cargo/git/db/bottlerocket-settings-sdk-3ffd24dec00e96c2

Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)

    Blocking waiting for file lock on package cache
   Compiling metal-dev v0.1.0 (/home/builder/actions-runner/_work/src/src/platform/base/bottlerocket/variants/metal-dev)
```

As alluded in other PRs and comments, I am wrapping twoliter in Bazel and in the variant build step I launch several variant builds (out of the same folder). This worked before the change to vendoring into `.cargo` - do you have any ideas on how to address it? I am a bit stumped as I think this is a result of the format of the `.cargo` folder with the vendor folder vs how it was before and the error seems to be entirely internal to `cargo`.

--- a/twoliter/embedded/Makefile.toml
+++ b/twoliter/embedded/Makefile.toml
@@ -362,13 +362,31 @@ for ws in sources variants .; do
   [ -s "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.lock" ] || continue
   CARGO_MANIFESTS+=(--sync "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml")
 done
-cargo vendor --locked --no-delete  --versioned-dirs --respect-source-config  ${CARGO_MANIFESTS[@]} .cargo/vendor > .cargo/config.toml 2> .cargo/vendor.log || {
-   echo "vendoring of cargo dependencies failed"
-   cat .cargo/vendor.log
-   exit 1
-}
-rm .cargo/vendor.log
+exec 9<>.cargo/vendor.lock
+if ! flock -w 90 9; then
+    echo "failed to obtain lock" >&2
+    exit 1
+fi
 
+set +e
+cargo vendor \
+    --locked \
+    --no-delete \
+    --versioned-dirs \
+    --respect-source-config \
+    ${CARGO_MANIFESTS[@]} \
+    .cargo/vendor \
+    > .cargo/twoliter_cargo_config.toml \
+    2> .cargo/vendor.log
+
+if [ "${?}" -ne 0 ]; then
+   echo "vendoring of cargo dependencies failed" >&2
+   cat .cargo/vendor.log >&2
+   exit 1
+fi
+
+set -e
+rm .cargo/vendor.log
 chmod -R o+r ${CARGO_HOME}
 '''
 ]
@@ -459,7 +477,8 @@ done
 if ! docker run --rm \
    -u $(id -u):$(id -g) \
    -e CARGO_HOME="/tmp/.cargo" \
-   -v "${CARGO_HOME}":/tmp/.cargo \
+   -v "${CARGO_HOME}/twoliter_cargo_config.toml":/tmp/.cargo/config.toml \
+   -v "${CARGO_HOME}/vendor":/tmp/.cargo/vendor \
    -v "${BUILDSYS_ROOT_DIR}/sources":/tmp/sources \
    "${TLPRIVATE_SDK_IMAGE}" \
    cargo fmt \
@@ -495,7 +514,8 @@ export VARIANT="${BUILDSYS_VARIANT}"
 if ! docker run --rm \
    -u $(id -u):$(id -g) \
    -e CARGO_HOME="/tmp/.cargo" \
-   -v "${CARGO_HOME}":/tmp/.cargo \
+   -v "${CARGO_HOME}/twoliter_cargo_config.toml":/tmp/.cargo/config.toml \
+   -v "${CARGO_HOME}/vendor":/tmp/.cargo/vendor \
    -v "${BUILDSYS_ROOT_DIR}/sources":/tmp/sources \
    -e VARIANT \
    "${TLPRIVATE_SDK_IMAGE}" \
@@ -978,7 +998,8 @@ docker run --rm \
   --user "$(id -u):$(id -g)" \
   --security-opt="label=disable" \
   -e CARGO_HOME="/tmp/.cargo" \
-  -v "${CARGO_HOME}":/tmp/.cargo \
+  -v "${CARGO_HOME}/twoliter_cargo_config.toml":/tmp/.cargo/config.toml \
+  -v "${CARGO_HOME}/vendor":/tmp/.cargo/vendor \
   -v "${BUILDSYS_ROOT_DIR}/sources":/tmp/sources \
   "${TLPRIVATE_SDK_IMAGE}" \
   bash -c "${run_cargo_deny}"

--- a/twoliter/embedded/build.Dockerfile
+++ b/twoliter/embedded/build.Dockerfile
@@ -112,7 +112,7 @@ USER builder
     # We only mount the config.toml and the vendor directory instead of the whole .cargo directory
     # because the .git and .registry directories are specific to the version of cargo and can cause
     # incapatibility between the host cargo and the cargo in the sdk
-RUN --mount=source=.cargo/config.toml,target=/home/builder/rpmbuild/.cargo/config.toml \
+RUN --mount=source=.cargo/twoliter_cargo_config.toml,target=/home/builder/rpmbuild/.cargo/config.toml \
     --mount=source=.cargo/vendor,target=/home/builder/rpmbuild/.cargo/vendor \
     --mount=type=cache,target=/home/builder/.cache,from=cache,source=/cache \
     --mount=source=sources,target=/home/builder/rpmbuild/BUILD/sources \

